machine/v1beta1/types_awsprovider.go

```diff
@@ -84,6 +84,13 @@ type AWSMachineProviderConfig struct {
 	// When omitted, no placement group is used when creating the EC2 instance.
 	// +optional
 	PlacementGroupName string `json:"placementGroupName,omitempty"`
+	// PlacementGroupPartition is the partition number within the placement group in which to launch the instance.
+	// This must be an integer value between 1 and 7. It is only valid if the placement group, referred in
+	// `PlacementGroupName` was created with strategy set to partition.
+	// +kubebuilder:validation:Minimum:=1
+	// +kubebuilder:validation:Maximum:=7
+	// +optional
+	PlacementGroupPartition int64 `json:"placementGroupPartition,omitempty"`
```

Comment:
Do you think we should add some CEL Validation? 

```go
!(has(self.placementGroupPartition) && !(has(self.placementGroupName) && size(self.placementGroupName)>0))

```
`message="placementGroupPartition is invalid, because placementGroupName is not set"`

Reply:
You can't use CEL in this API because it is a raw extension, but, for upstream you could try to get that through yes